main.js

```diff
@@ -476,60 +479,86 @@ function createWindow() {
-      console.log(msg);
-      mainWindow.send('updateNotAvailable');
-    });
+  autoUpdater.on('checking-for-update', () => {
+    mainWindow.send('updateChecking');
+    mainWindow.send('consoleMsg', `Checking for update at ${autoUpdater.getFeedURL()}`);
+  });
 
-    autoUpdater.on('update-available', () => {
-      mainWindow.send('updateAvailable');
-    });
+  autoUpdater.on('error', (err, msg) => {
+    console.log('update error');
+    console.log(msg);
+    mainWindow.send('consoleMsg', msg);
+    mainWindow.send('error', msg);
+  });
 
-    autoUpdater.on('update-downloaded', (e, releaseNotes, releaseName,
-      releaseDate, updateUrl, quitAndUpdate) => {
-      // Old way of doing things
-      // mainWindow.webContents.executeJavaScript('$(".js-softwareUpdate")
-      // .removeClass("softwareUpdateHidden");');
-      console.log(quitAndUpdate);
-      mainWindow.send('updateReadyForInstall');
-    });
+  autoUpdater.on('update-not-available', () => {
+    console.log('update not available');
+    mainWindow.send('updateNotAvailable');
+    mainWindow.send('consoleMsg', 'Update Not Available');
+  });
 
-    // Listen for installUpdate command to install the update
-    ipcMain.on('installUpdate', () => {
-      autoUpdater.quitAndInstall();
-    });
+  autoUpdater.on('update-available', () => {
+    console.log('update available');
+    mainWindow.send('updateAvailable');
+    mainWindow.send('consoleMsg', 'Update Available');
+  });
 
-    // Listen for checkForUpdate command to manually check for new versions
-    ipcMain.on('checkForUpdate', () => {
-      autoUpdater.checkForUpdates();
-    });
+  autoUpdater.on('download-progress', (e, data) => {
+    console.log('Update download progress');
+    console.log(data);
+    mainWindow.send('consoleMsg', data);
+  });
+
+  autoUpdater.on('update-downloaded', (e, releaseNotes, releaseName,
+                                       releaseDate, updateUrl) => {
+    console.log('update ready for install');
+    console.log(releaseNotes);
+    console.log(releaseName);
+    console.log(releaseDate);
+    console.log(updateUrl);
+    const opts = {};
+    opts.Name = releaseName;
+    opts.URL = updateUrl;
+    opts.Date = releaseDate;
+    opts.Notes = releaseNotes;
+    mainWindow.send('updateReadyForInstall', opts);
+    mainWindow.send('consoleMsg', 'Update Ready for Install');
+    mainWindow.send('consoleMsg', opts);
+  });
+
+// Listen for installUpdate command to install the update
+  ipcMain.on('installUpdate', () => {
+    console.log('Quit and Install Update');
+    autoUpdater.quitAndInstall();
+  });
+
+// Listen for checkForUpdate command to manually check for new versions
+  ipcMain.on('checkForUpdate', () => {
+    autoUpdater.checkForUpdates();
+  });
 
-    autoUpdater.setFeedURL(feedURL);
+  autoUpdater.setFeedURL(feedURL);
 
+  mainWindow.webContents.on('dom-ready', () => {
     // Check for updates every hour
     autoUpdater.checkForUpdates();
```

Comment:
Could we make a separate checkForUpdates function which would reset the interval? Otherwise, this is checking every hour irrespective of any manual checks from the client. So, if the user manually checks 1 minute before the interval is due up, then there would needlessly be a second check a minute after a previous one.

Something like this should work:

```
let checkForUpdatesInterval;
const checkForUpdates = () => {
  clearInterval(checkForUpdatesInterval);
  autoUpdater.checkForUpdates();
  checkForUpdatesInterval = setInterval(() => {
    autoUpdater.checkForUpdates();
  }, 60 * 60 * 1000);
}
```

then anywhere you are calling `autoUpdater.checkForUpdates();`, call `checkForUpdates()` instead.